adr/ADR-44.md

```diff
@@ -177,8 +182,17 @@ the meta leader version is known - but in reality one cannot really tell a lot f
 guaranteed to be representative of the cluster and is particularly problematic in long running clients as the server
 may have been upgraded since last `INFO` call.
 
-In the future we could have clients assert that a certain API call requires a certain API Level but it was felt that
-today that would not be feasible to do given the amount of clients and their current state.
+# Required API level
+
+Clients can assert that a certain API call requires a minimum API Level by including a `Nats-Required-Api-Level` header,
+containing the requested minimum API level as a string. All endpoints under the JetStream API (`$JS.API.>`) should be
+supported.
+
+This asserts whether the server that's responding supports this API level. The request will error with
+`api level not supported` if the server has a lower API level than required.
+
+A single server does not represent a cluster-wide supported API level. In the future we should keep track of a
+cluster-agreed API level, and have the servers enforce an agreed-upon API level.
```

Comment:
Seems good, curious about thoughts from @Jarema and @aricart

Reply:
Perhaps when `Nats-Required-Api-Level` is used for the request, we could have the server that's responding include a `Nats-Api-Level` header containing the highest supported API level?
In that case, if the client uses the request header but doesn't get the response header, it knows it's an older server.

Reply:
This is a good idea, just conscious of the big increase in network traffic - and for some requests like pulls this would still be quite hard to make sense of as those dont directly respond to the API request

Need to think a bit on this aspect

Reply:
As discussed, we'll leave just the `Nats-Required-Api-Level` on the request for 2.12
And we'll revisit and improve later once we're closer to having a cluster-agreed API level, feature flags, etc.